projects/optic/src/commands/capture/capture.ts

```diff
@@ -492,3 +450,145 @@ function makeRequests(
     );
   });
 }
+
+async function runRequestsCommand(
+  command: string,
+  proxyVar: string,
+  proxyUrl: string
+): Promise<void> {
+  const cmd = commandSplitter(command);
+  process.env[proxyVar] = proxyUrl;
+  const reqCmd = spawn(cmd.cmd, cmd.args, {
+    detached: true,
+    shell: true,
+  });
+
+  let reqCmdPromise: Promise<void>;
+  reqCmdPromise = new Promise((resolve, reject) => {
+    reqCmd.on('exit', (code) => {
+      if (code === 0) {
+        resolve();
+      } else {
+        reject();
+      }
+    });
+  });
+  reqCmd.stderr.on('data', (data) => {
+    logger.error(data.toString());
+  });
+  return reqCmdPromise;
+}
+
+function makeAllRequests(
+  captureConfig: CaptureConfigData,
+  proxy: ProxyInstance
+) {
+  // send requests
+  let sendRequestsPromise: Promise<any> = Promise.resolve();
+  if (captureConfig.requests) {
+    const requests = sendRequests(
+      captureConfig.requests,
+      proxy.url,
+      captureConfig.config?.request_concurrency || 5
+    );
+    sendRequestsPromise = Promise.allSettled(requests);
+  }
+
+  // run requests command
+  let runRequestsPromise: Promise<void> = Promise.resolve();
+  if (captureConfig.requests_command) {
+    const proxyVar =
+      captureConfig.requests_command.proxy_variable || 'OPTIC_PROXY';
+
+    runRequestsPromise = runRequestsCommand(
+      captureConfig.requests_command.command,
+      proxyVar,
+      proxy.url
+    );
+  }
+
+  return [sendRequestsPromise, runRequestsPromise];
+}
+
+async function serverReady(
+  checkUrl: string,
+  interval: number,
+  timeout: number
+) {
+  const now = Date.now();
+  const spinner = ora('Waiting for server to come online...');
+  spinner.start();
+  spinner.color = 'blue';
+
+  const checkServer = (): Promise<boolean> =>
+    fetch(checkUrl)
+      .then((res) => String(res.status).startsWith('2'))
+      .catch((e) => {
+        logger.debug(e);
+        return false;
+      });
+
+  let done = false;
+  while (!done) {
+    const isReady = await checkServer();
+    if (isReady) {
+      spinner.succeed('Server check passed');
+      done = true;
+    } else if (Date.now() > now + timeout) {
+      throw new UserError('Server check timed out.');
+    }
+    await wait(interval);
+  }
+}
+
+async function specExists(filePath: string): Promise<boolean> {
+  try {
+    await fs.stat(filePath);
+  } catch {
+    return false;
+  }
+  return true;
+}
```

Comment:
nit: I think this is an unnecessary layer of indirection - we can just put it in the `setup` function since it's only called once, and the setup fn is fairly straightforward